Improve ControlBlock background drawing

dylanmccall opened the pull request:
Instead of reusing the background.gd script to draw a gutter in ControlBlock, define a new script (gutter.gd) specifically for that purpose. In the existing background.gd script, draw a left border which is expected to flow into the gutter if shift_top or shift_bottom is set.

https://phabricator.endlessm.com/T35538

dbnicholson commented on addons/block_code/ui/blocks/control_block/control_block.gd:
```diff
@@ -148,9 +143,18 @@ func format():
+
+		var snap_gutter := Control.new()
+		snap_gutter.name = "Background"
+		snap_gutter.set_script(preload("res://addons/block_code/ui/blocks/utilities/background/gutter.gd"))
```
I've seen this in a couple places in here. Why not just instantiate the class directly?

dylanmccall replied:
I'm not sure _why_ we don't do that, indeed. I'll fix that :)

manuq replied:
Yep, `set_script()` only makes sense in the BlockCode node when it replaces the parent script. All others should be fixed.

manuq reviewed (changes requested):
It looks very nice! The `set_script` that was there before for other backgrounds and should be changed. Besides that, please leave the Pong scene out of this PR.

dylanmccall commented:
In the newest set of commits, I removed the unnecessary change to `pong_game.tscn` and I replaced that `set_script()` scheme with instantiating `BlockBackground` and `BlockGutter` classes directly.

dbnicholson commented on addons/block_code/ui/blocks/control_block/control_block.gd:
```diff
@@ -144,9 +143,8 @@ func format():
 		snap_container.size_flags_horizontal = Control.SIZE_SHRINK_BEGIN
 
-		var snap_gutter := Control.new()
+		var snap_gutter := BlockGutter.new()
```
I think we've been trying to move away from global classes unless their really needed since it reduces the chance for startup issues with the global class cache. Instead, I think you should just preload them at the top:

```
const BlockBackground = preload("res://addons/block_code/ui/blocks/utilities/background/background.gd")
const BlockGutter = preload("res://addons/block_code/ui/blocks/utilities/background/gutter.gd")
```

manuq replied:
Yes, exactly. Can you make this last change @dylanmccall ?

dylanmccall replied:
Ah, I missed that happening! Makes sense :) That's fixed in the newest commit.

dbnicholson replied:
I think Manuel started it near the beginning of GUADEC in the context of [import errors](https://phabricator.endlessm.com/T35494) and I happened to notice it since I was trying to wrap up some of my own work that Thursday and Friday.

dbnicholson reviewed (changes requested):
I can't say that I really followed the polygon changes, but I trust their right. I just had one small suggestion on class names.

manuq commented on addons/block_code/ui/blocks/utilities/background/background.gd:
```diff
@@ -1,4 +1,5 @@
 @tool
+class_name BlockBackground
```
As Dan pointed, please remove `class_name` and use preload instead.

dbnicholson reviewed (approved):
Lovely, thanks!
